### PR TITLE
[LEVWEB-1396] Add redaction of civil partnerships

### PR DESCRIPTION
**Pull request by aarongill1**

### CODE REVIEW

File: src/lib/censorRecords.js [hunk 1]
```diff
@@ -69,8 +69,8 @@ const censorDeath = (death, roles) => {
       marginalNote: censorField(false, death.status.marginalNote),
       onAuthorityOfRegistrarGeneral: censorField(true, death.status.onAuthorityOfRegistrarGeneral)
     },
-    previousRegistration: death.previousRegistration && censorRecords(death.previousRegistration, roles),
-    nextRegistration: death.nextRegistration && censorRecords(death.nextRegistration, roles),
+    previousRegistration: death.previousRegistration && censorDeath(death.previousRegistration, roles),
+    nextRegistration: death.nextRegistration && censorDeath(death.nextRegistration, roles),
```
Comment (daniel-ac-martin) [inline src/lib/censorRecords.js:73]: Was this a bug that we missed?

File: src/lib/censorRecords.js [hunk 2]
```diff
@@ -205,4 +205,95 @@ const censorMarriage = (marriage, roles) => {
...
+        sex: censorField(true,partnership.partner2.sex),
+        aliases: censorField(true,partnership.partner2.aliases) || [],
+        occupation: censorField(true,partnership.partner2.occupation),
+        retired: censorField(true,partnership.partner2.retired),
+        condition: censorField(true,partnership.partner2.condition),
+        signature: censorField(true,partnership.partner2.signature)
+      },
+     fatherOfPartner1: {
+        surname: censorField(true,partnership.fatherOfPartner1.surname),
+        forenames: censorField(true,partnership.fatherOfPartner1.forenames),
```
Comment (daniel-ac-martin) [inline src/lib/censorRecords.js:250]: Funny indentation here.

Comment (aarongill1) [reply src/lib/censorRecords.js:250]: Fixed

File: test/unit/lib/censorPartnership.js [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+const { censorPartnership } = require('../../../src/lib/censorRecords');
+const roleWithFullAccess = ['full-details'];
+const roleWithoutFullAccess = ['births'];
+
+const partnershipData = {
+  id: 3000,
+    dateOfPartnership: "2020-09-23",
+  placeOfPartnership: {
+  address: "Address of Place of Partnership",
+    short: false
```
Comment (daniel-ac-martin) [inline test/unit/lib/censorPartnership.js:11]: Funny indentation.

`short` should be a string. (Can be just "Place of Partnsership".)

File: test/unit/lib/censorPartnership.js [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+      surname: "ONE",
+      suffix: "ESQ",
+      address: "PARTNER ONE ADDRESS",
+      dob: "1987-09-13",
+      sex: "Male",
+      aliases: [],
+      occupation: "Hotelier",
+      retired: false,
+      condition: "Single",
+      signature: "Partner 1 Signature"
```
Comment (daniel-ac-martin) [inline test/unit/lib/censorPartnership.js:28]: Indentation.

File: test/unit/lib/censorPartnership.js [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+  },
+  witness2: {
+    signature: "WITNESS 2 SIGNATURE",
+      signatureIsMark: true
+  },
+  status: {
+    blocked: false,
+      correction: "CORRECTION",
+      marginalNote: "MARGINAL NOTE",
+      onAuthorityOfRegistrarGeneral: "AUTHORITAH!"
```
Comment (daniel-ac-martin) [inline test/unit/lib/censorPartnership.js:88]: Are these definitely strings?

File: test/unit/lib/censorPartnership.js [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+  previousRegistration: undefined,
+    nextRegistration: undefined
+}
+
+const redactedPartnershipDataNotBlocked = {
+  id: 3000,
+  dateOfPartnership: "2020-09-23",
+  placeOfPartnership: {
+    address: "Address of Place of Partnership",
+    short: false
```
Comment (daniel-ac-martin) [inline test/unit/lib/censorPartnership.js:99]: Needs to be a string.

File: test/unit/lib/censorPartnership.js [hunk 7]
```diff
@@ -0,0 +1,304 @@
+'use strict';
+const { censorPartnership } = require('../../../src/lib/censorRecords');
+const roleWithFullAccess = ['full-details'];
+const roleWithoutFullAccess = ['births'];
+
+const partnershipData = {
```
Comment (daniel-ac-martin) [inline test/unit/lib/censorPartnership.js:6]: Now that I think of it...

I wonder if we should make use of https://github.com/UKHomeOffice/lev-api/blob/master/mock/partnership_registration_v1.json

Forgot we had that! :-/

**Review by daniel-ac-martin — CHANGES_REQUESTED** [review]
Spotted a few things.

File: test/unit/lib/censorPartnership.js (hunk 3 above)
Comment (aarongill1) [reply test/unit/lib/censorPartnership.js:11]: fixed

File: test/unit/lib/censorPartnership.js (hunk 6 above)
Comment (aarongill1) [reply test/unit/lib/censorPartnership.js:99]: Fixed

File: test/unit/lib/censorPartnership.js (hunk 5 above)
Comment (aarongill1) [reply test/unit/lib/censorPartnership.js:88]: onAuthorityOfRegistrarGeneral is a boolean. Investigation shows correction and onAuthority... are not required for partnership and marriage records, so removed fields in respective censorRecord functions and also tests.

File: test/unit/lib/censorPartnership.js (hunk 7 above)
Comment (aarongill1) [reply test/unit/lib/censorPartnership.js:6]: The data is in a different 'shape' and doesn't have any blocked records - I think the work required to refactor would be more than it's worth.

File: src/lib/censorRecords.js (hunk 1 above)
Comment (aarongill1) [reply src/lib/censorRecords.js:73]: Not sure I get what you're asking but it was originally removed due to a build failure, refactored but then it turned out to be a different problem and was subsequently added back.

Comment (daniel-ac-martin) [reply src/lib/censorRecords.js:73]: Well there is no `censorRecords` function is there? So the old code would crash in the presence of a chain and this is a bug-fix that we should publish ASAP.

**Pull request merged**